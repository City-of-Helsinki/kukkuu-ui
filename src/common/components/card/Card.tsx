import React, { FunctionComponent, ReactNode, ReactElement } from 'react';
import { Button } from 'hds-react';

import angleDownIcon from '../../../assets/icons/svg/angleDown.svg';
import styles from './card.module.scss';
import Icon from '../icon/Icon';

interface CardProps {
  action: () => void;
  actionText: string;
  alt?: string;
  children: ReactNode;
  imageElement?: ReactElement;
  focalContent?: ReactNode;
  imageSrc?: string;
  primaryAction?: () => void;
  primaryActionText?: string;
  title: string;
}

const Card: FunctionComponent<CardProps> = ({
  action,
  actionText,
  alt = '',
  children,
  imageElement,
  focalContent,
  imageSrc,
  primaryAction,
  primaryActionText,
  title,
}) => {
  return (
    <div
      className={styles.wrapper}
      onClick={primaryAction ? primaryAction : action}
    >
      <div className={styles.start}>
        {imageSrc ? (
          <img src={imageSrc} alt={alt} className={styles.image} />
        ) : (
          imageElement && imageElement
        )}
      </div>

      <div className={styles.middle}>
        <h3 className={styles.title}>{title}</h3>
        <div className={styles.focalPoint}>
          {primaryAction && (
            <Button className={styles.primaryActionButton}>
              {primaryActionText}
            </Button>
          )}
          {focalContent && focalContent}
        </div>
        {children}
      </div>

      <div className={styles.end}>
        <button aria-label={actionText} className={styles.actionWrapper}>
          <div className={styles.actionText}>{actionText}</div>
          <Icon src={angleDownIcon} alt={''} className={styles.gotoAction} />
        </button>
      </div>
    </div>
  );
};

export default Card;
