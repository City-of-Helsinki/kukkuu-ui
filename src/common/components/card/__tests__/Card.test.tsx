import { render } from '../../../test/testingLibraryUtils';
import Card from '../Card';

it('renders snapshot correctly', () => {
  const { container } = render(
    <Card action={jest.fn()} actionText={''} title={''}>
      foo
    </Card>
  );
  expect(container).toMatchSnapshot();
});
