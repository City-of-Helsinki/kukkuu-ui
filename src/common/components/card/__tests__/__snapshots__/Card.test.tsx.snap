// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `"<div class=\\"wrapper\\"><div class=\\"start\\"></div><div class=\\"middle\\"><h3 class=\\"title\\"></h3>foo<div class=\\"focalPoint\\"></div></div><div class=\\"end\\"><button type=\\"button\\" class=\\"Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO actionWrapper\\" style=\\"--color:var(--color-black);--color-hover:var(--color-black);--color-focus:var(--color-black-90);--color-hover-focus:var(--color-black-90);--background-color:transparent;--background-color-hover:transparent;--background-color-focus:transparent;--background-color-hover-focus:transparent;--background-selected:var(--color-summer);--border-color:none;--border-color-hover:var(--color-summer-dark-50);--border-color-focus:var(--color-summer-dark-50);--border-color-hover-focus:var(--color-summer-dark-50);--border-color-selected:var(--color-summer);--border-color-selected-hover:var(--color-summer-dark-50);--border-color-selected-focus:var(--color-summer);--border-color-selected-hover-focus:var(--color-summer)\\"><span class=\\"Button-module_label__a4np1 button_hds-button__label__2EQa-\\"><div class=\\"actionText\\"></div><div class=\\"imgWrapper gotoAction\\"><img src=\\"angleDown.svg\\" alt=\\"\\"/></div></span></button></div></div>"`;
